clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -475,7 +475,9 @@ private static int parseAcks(String acksString) {
     public void initTransactions() {
         if (transactionManager == null)
             throw new IllegalStateException("Cannot call initTransactions without setting a transactional id.");
-        transactionManager.initializeTransactions().get();
+        TransactionalRequestResult result = transactionManager.initializeTransactions();
+        sender.wakeup();
```

Comment:
This is just a safety. We need the wakeup in the other methods since the client may be in a poll with a very timeout. However,  `initTransactions` should be the first method called, even before a send (we actually enforce this elsewhere), so there is unlikely to be a poll with a long timeout.